internal/osutils/shortcut/shortcut_linux.go

```diff
@@ -0,0 +1,116 @@
+package shortcut
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+)
+
+type Shortcut struct {
+	target string
+	path   string
+}
+
+func New(target, path string) (*Shortcut, error) {
+	if !fileutils.FileExists(target) {
+		return nil, errs.New("Target does not exist")
+	}
+
+	scut := Shortcut{
+		target: target,
+		path:   path,
+	}
+
+	return &scut, nil
+}
+
+type ShortcutSaveOpts struct {
+	GenericName string
+	Comment     string
+	Keywords    string
+	IconData    []byte
+	IconPath    string
+}
+
+func (s *Shortcut) Save(name string, opts ShortcutSaveOpts) error {
+	t := template.New("")
+	t, err := t.Parse(desktopFileTmpl)
+	if err != nil {
+		return errs.Wrap(err, "Could not parse desktop file template")
+	}
```

Comment:
fyi; http://github.com/ActiveState/cli/blob/9faf61f1e772249d29da4b95fadae575faf1c7a5/internal/strutils/strutils.go#L17-L17

Reply:
The params type required data to be structured more magically than necessary, so I swapped from map[string]interface{} to interface{}. Other callers should not be affected by this change.